conda/common/logic.py

```diff
@@ -330,9 +330,21 @@ def _assign_no_name(self, vals):
 
     def Convert_(self, x):
         tx = type(x)
+        name = x
         if tx in (tuple, list):
             return tx(map(self.Convert_, x))
-        return self.names.get(x, x)
+        elif tx not in (bool, int) and name not in self.names:
```

Comment:
Hi @msarahan,
I'm working on the preparation part of https://github.com/curiu/notes/issues/2, i.e., cleaning up this module.
I don't understand this change. Can you elaborate on the need for this?
Ideally, `Convert_` would
- leave `True`, `False` and valid `int`s (i.e., `0 < abs(x) <= self.m`) untouched,
- convert registered names into the corresponding valid values,
- error out in case of invalid inputs.

Before this change, it would silently accept wrong inputs and return them as-is which would likely lead to raising some unexpected errors shortly after.
After this change, it silently accepts wrong inputs and adds corresponding new variables for non-mapped names.

Both behaviors, before and after, are wrong, so this change didn't make things much worse ;). It might make things fail more silently, though.
Do you have any test case that lead to this change?

Reply:
I am certain that you have a better handle on this than I did/do. I almost certainly considered silent falling as the improvement I needed without realizing that was what was happening.

For a test case, I recommend just reverting this bit and seeing what failures show up. I unfortunately don't remember any more.

Reply:
> I unfortunately don't remember any more.

Ah, bummer -- but totally understandable ;). Thanks for the prompt answer!